# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/storage/SessionStorageManager.h
# Commit: 2663311a1b62
# Full Hash: 2663311a1b628c835b7c557d453797080c92d25d
# Author: Yaron Tausky <ytausky@mozilla.com>
# Date: 2019-12-09 21:50:19
# Regressor Bug: 1593246
# File Overlap Count: 1
# Description:
#   Bug 1593246 - Part 2: Give SessionStorageManager a reference to BrowsingContext r=sg
#   
#   This reference is necessary when sending session storage data for
#   all browsing context to the parent process. Note that it entails
#   making SessionStorageManager a cycle collection participant, since
# ==============================================================================

diff -r d9f0d827e28d -r 2663311a1b62 dom/storage/SessionStorageManager.h
--- a/dom/storage/SessionStorageManager.h	Mon Dec 09 13:04:51 2019 +0000
+++ b/dom/storage/SessionStorageManager.h	Thu Dec 05 17:51:59 2019 +0000
@@ -21,12 +21,18 @@
 class SessionStorageManager final : public nsIDOMSessionStorageManager,
                                     public StorageObserverSink {
  public:
-  SessionStorageManager();
+  explicit SessionStorageManager(RefPtr<BrowsingContext> aBrowsingContext);
 
-  NS_DECL_ISUPPORTS
+  NS_DECL_CYCLE_COLLECTING_ISUPPORTS
   NS_DECL_NSIDOMSTORAGEMANAGER
   NS_DECL_NSIDOMSESSIONSTORAGEMANAGER
 
+  NS_DECL_CYCLE_COLLECTION_CLASS(SessionStorageManager)
+
+  RefPtr<BrowsingContext> GetBrowsingContext() const {
+    return mBrowsingContext;
+  }
+
  private:
   ~SessionStorageManager();
 
@@ -55,6 +61,8 @@
   nsClassHashtable<nsCStringHashKey, OriginKeyHashTable> mOATable;
 
   RefPtr<SessionStorageObserver> mObserver;
+
+  RefPtr<BrowsingContext> mBrowsingContext;
 };
 
 }  // namespace dom